src/test/java/com/facebook/openwifirrm/optimizers/TestUtils.java

```diff
@@ -644,7 +696,77 @@ public static State createState(
 			new int[] { channelWidthA, channelWidthB },
 			new int[] { txPowerA, txPowerB },
 			new String[] { bssidA, bssidB },
-			new int[][] { clientRssisA, clientRssisB }
+			new String[][] {},
+			new int[][] { clientRssisA, clientRssisB },
+			DEFAULT_LOCAL_TIME
+		);
+	}
+
+	public static AggregatedState createAggregatedState(
+		int channel,
+		int channelWidth,
+		int txPower,
+		String bssid,
+		String station,
+		int[] clientRssi
+	) {
+		AggregatedState state = new AggregatedState();
+		state.radio = new AggregatedState.Radio(channel, channelWidth, txPower);
+		state.bssid = bssid;
+		state.station = station;
+		for (int rssi : clientRssi) {
+			state.rssi.add(rssi);
+		}
+		return state;
+	}
+
+	/**
+	* Create a device state object with two radios.
+	*
+	* @param channelA channel number
+	* @param channelWidthA channel width (MHz) of channelA
+	* @param txPowerA tx power (dB) for channelA
+	* @param bssidA bssid for radio on channelA
+	* @param clientRssisA array of client RSSIs for channelA
+	* @param channelB channel number
+	* @param channelWidthB channel width (MHz) of channelB
+	* @param txPowerB tx power (dB) for channelB
+	* @param bssidB bssid for radio on channelB
+	* @param clientRssisB array of client RSSIs for channelB
+	* @param localtime local time for the State
+	* @return the state of an AP with two radios
+	*/
+	public static State createState(
+		int channelA,
+		int channelWidthA,
+		int txPowerA,
+		String bssidA,
+		String[] stationsA,
+		int[] clientRssisA,
+		int channelB,
+		int channelWidthB,
+		int txPowerB,
+		String bssidB,
+		String[] stationsB,
+		int[] clientRssisB,
+		long localtime
+	) {
+		return createState(
+			new int[] { channelA, channelB },
+			new int[] { channelWidthA, channelWidthB },
+			new int[] { txPowerA, txPowerB },
+			new String[] { bssidA, bssidB },
+			new String[][] { stationsA, stationsB },
+			new int[][] { clientRssisA, clientRssisB },
+			localtime
+		);
+	}
+
+	public static String getBssidStationKeyPair(String bssid, String station) {
```

Comment:
Since you do have a method for this, maybe move it somewhere else (`ModelerUtils`? Or `UCentralUtils`?) and use it in `ModelerUtils` instead of `String.format` there?